portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```
@@ -14,19 +14,85 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.lang.ClassCastException;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  ArrayList<String> employeeComment = new ArrayList<String>();
+
+    /**
+   * Get the number of comments to display from the user
+   */
+  private int getNumComments(HttpServletRequest request) {
+    // Get the input from the form.
+    String userNum = request.getParameter("number-comments");
+    int requestedNum;
+    //Convert input into an int
+    try {
+      requestedNum = Integer.parseInt(userNum);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + userNum);
```

I definitely appreciate the error handling! But instead of using print statments, let's throw an exception (perhaps an IllegalArgumentException here). In general, we want to code _defensively_, which means we want to fail requests that are poorly formed, or don't conform to our expectations. Throwing an exception is a good way to indicate that to the calling user (whether that's Javscript that you write, or an accessor of your API).

---

Can you do that throughout? IllegalArgumentExceptions are a good semantic choice in many cases (where you're validating the input of a user), but so are IllegalStateExceptions (when you're validating an assumption you have about what a function or datastore will contain/return)